packages/@aws-cdk/core/lib/app.ts

```diff
@@ -87,9 +87,33 @@ export class App extends Construct {
     return APP_SYMBOL in obj;
   }
 
+  /**
+   * Return the root App of the given Construct
+   *
+   * Throws if the construct is not in an App.
+   */
+  public static of(construct: IConstruct): App {
+    const root = construct.node.root;
+    if (!App.isApp(root)) {
+      throw new Error(`Construct '${construct.node.path}' should be created under an 'App'`);
+    }
+    return root;
+  }
+
+  /**
+   * Accessor for a field that's private to App
+   *
+   * Get the Cloud Assembly output directory.
+   *
+   * @experimental
+   */
+  public static outdirOf(obj: App): string {
```

Comment:
Why a static method that takes `App` rather than modify the instance variable `outdir` to be a public getter?